### ISSUE TITLE
Cannot run fluxgui on kali linux: "ImportError: No module named gconf"

### ISSUE THREAD

**Issue by RJ151** [xflux-gui/fluxgui#90]
Apt-get install python-gconf has no installation candidate on kali-rolling.
Searched for solution and cannot find one- output of fluxgui is below. 

```
root@kali:/tmp/fluxgui# fluxgui 
Traceback (most recent call last):
  File "/usr/local/bin/fluxgui", line 17, in <module>
    from fluxgui.fluxapp import main
  File "/usr/local/lib/python2.7/dist-packages/fluxgui/fluxapp.py", line 3, in <module>
    from fluxgui import fluxcontroller, settings
  File "/usr/local/lib/python2.7/dist-packages/fluxgui/fluxcontroller.py", line 1, in <module>
    from fluxgui import xfluxcontroller
  File "/usr/local/lib/python2.7/dist-packages/fluxgui/xfluxcontroller.py", line 5, in <module>
    from fluxgui import settings
  File "/usr/local/lib/python2.7/dist-packages/fluxgui/settings.py", line 2, in <module>
    import gconf
ImportError: No module named gconf

```

**Comment by faraco** [comment]
Hi there. I believe this is likely Kali Linux repository issue. Do you mind to give a bit more details especially your Kali Linux release you are using?

PS: Running this command will print your distribution release:  `lsb_release -a | awk '/Release/ {print $2}'`

**Comment by RJ151** [comment]
root@kali: lsb_release -a | awk '/Release/ {print $2}'
No LSB modules are available.
kali-rolling

**Comment by mpierret** [comment]
Hi!
I have the same issue on Debian buster. The package is not available : https://packages.debian.org/sid/python-gconf

**Comment by faraco** [comment]
It seems these GNU/Linux distributions releases are newer. The potential reason why there are not distributing python-gconf in the newer version probably due to the GConf deprecation and the distribution community preference of DConf/GSettings  + GTK 3 environment (just assuming).

python-gconf availability for Debian (or derived) stops at Debian Stretch - https://pkgs.org/download/python-gconf

I believe it is still possible to build & install them yourself if you still want to use xflux-gui on your new GNU/Linux install, however.

**Comment by mpierret** [comment]
I'm really new to the linux world, so this may be irrelevant, but doesn't the fact that the package is available for sid/unstable suggest that it should be available for buster/testing quite soon ?

**Comment by ntc2** [comment]
As @faraco mentions, GConf is deprecated. User @bil-elmoussaoui did some work on porting `fluxgui` to DConf last year, but he never sent a PR. Related discussion in #34 and #48.

**Comment by faraco** [comment]
If gio.Settings available for Python 2, then migration from gconf to dconf would be a breeze.

Why don't we just store and retrieve data from typical JSON/YAML/other-config-files instead? Eliminating dconf as a dependency for both python2 + gtk2 (or in future, python3 + gi)?

AFAIK, the only problem stopping this idea from being implemented is both gconf and dconf act as a daemon to auto-apply the configuration data while fluxgui running in the background which plainly getting and setting data via JSON for instance and waiting for user input probably won't work properly and even inconvenient.

However, it might be possible to implement a daemon to auto-apply the configuration using python-daemon module.

Anyway, I do think that migrating to Gtk+ 3 might be an overall a good idea.

I'm planning to start my own fork of fluxgui for Gtk+ 3 migration. I know @bilelmoussaoui mentioned us that he is also working on the same plan. I just wish he at least can release the source code (if that was his intention in the first place) of the project progress so I can help to contribute anything to boost the development productivity.

**Comment by ntc2** [comment]
I just merged PR #112 which ports `fluxgui` to Python 3 and GTK+3. Please let me know if this issue should still be open.
